### PR TITLE
ENT-226: Replace conf.py with known-good version

### PR DESCRIPTION
**Pull request by mattdrayer**
This PR replaces the existing Sphinx "conf.py" file (which I could not get to work locally) with a modified version of a known-good configuration stolen from https://github.com/edx/course-discovery/blob/master/docs/conf.py.  

Using this file I was able to run the "make html" command in a local Python virtual environment (virtualenv) after running "pip install" using the "doc.txt" requirements file.  The end result was HTML docs in the "_build/html" directory.

I'm not sure if this is correct, but it does work, so I'm posting the changes here for you all to review.

@srpearce @grantgoodmanedX @e-kolpakov

### CODE REVIEW

File: docs/conf.bak [hunk 1]
```diff
@@ -0,0 +1,489 @@
+# -*- coding: utf-8 -*-
```
Comment (mattdrayer) [inline docs/conf.bak]: Note that this is just the original file -- I kept it around for reference -- if we are good with the changes I'll remove the file from the PR before merging.

**Comment by mattdrayer** [comment]
[](url)
screen shot 2017-02-16 at 12 52 21 am

**Comment by mattdrayer** [comment]
@bradenmacdonald too :)

File: docs/conf.py [hunk 2]
```diff
@@ -1,133 +1,103 @@
...
 # needs_sphinx = '1.0'
 
-# Add any Sphinx extension module names here, as strings. They can be
-# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
-# ones.
-extensions = [
-    'edx_theme',
-    'sphinx.ext.autodoc',
-    'sphinx.ext.doctest',
-    'sphinx.ext.intersphinx',
```
Comment (e-kolpakov) [inline docs/conf.py]: @mattdrayer intersphinx allows interlinking docs from different artifacts - probably we still need it.

File: docs/conf.py [hunk 3]
```diff
@@ -1,133 +1,103 @@
...
+project = u'Open edX Enterprise Service'
+copyright = u'2016-{year}, edX Inc.'.format(year=datetime.date.today().year)
 
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 #
 # The short X.Y version.
-version = enterprise.__version__
+version = '0.1'
```
Comment (e-kolpakov) [inline docs/conf.py]: @mattdrayer Not good. As you can see, everyone just forgets to update it. Please try your best to still keep it.

File: docs/conf.py [hunk 4]
```diff
@@ -136,349 +106,163 @@
...
-#
-# epub_show_urls = 'inline'
-
-# If false, no index is generated.
-#
-# epub_use_index = True
-
-
-# Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {
```
Comment (e-kolpakov) [inline docs/conf.py]: @mattdrayer Please try to keep intersphinx configs as well.

File: docs/conf.py [hunk 5]
```diff
@@ -136,349 +106,163 @@
...
 }
 
 # Grouping the document tree into LaTeX files. List of tuples
-# (source start file, target name, title,
-#  author, documentclass [howto, manual, or own class]).
+# (source start file, target name, title, author, documentclass [howto/manual]).
 latex_documents = [
-    (master_doc, 'edx-enterprise.tex', 'enterprise Documentation',
-     author, 'manual'),
+    ('index', 'enterprise.tex', u'Open edX Enterprise Service Documentation',
```
Comment (e-kolpakov) [inline docs/conf.py]: @mattdrayer nit: project + ' Documentation'?

File: docs/conf.py [hunk 6]
```diff
@@ -136,349 +106,163 @@
...
+     u'edX', 'manual'),
```
Comment (e-kolpakov) [inline docs/conf.py]: @mattdrayer nit: extract `author` variable:

    author = u'edX'
    latex_documents = [(..., author, 'maunal')]

File: docs/conf.py [hunk 7]
```diff
@@ -136,349 +106,163 @@
...
 
-# -- Options for manual page output ---------------------------------------
+# -- Options for manual page output --------------------------------------------
 
 # One entry per manual page. List of tuples
 # (source start file, name, description, authors, manual section).
 man_pages = [
-    (master_doc, 'enterprise', 'enterprise Documentation',
-     [author], 1)
+    ('index', 'Open edX Enterprise Service', u'Open edX Enterprise Service Documentation',
```
Comment (e-kolpakov) [inline docs/conf.py]: @mattdrayer nit:

    master_doc = 'index'
    docs_heading = "{project_name} Documentation".format(project_name=project)
    
    man_pages = [(master_doc, project, docs_heading, [author], 1)

File: docs/conf.py [hunk 8]
```diff
@@ -136,349 +106,163 @@
...
 # Grouping the document tree into Texinfo files. List of tuples
 # (source start file, target name, title, author,
 #  dir menu entry, description, category)
 texinfo_documents = [
-    (master_doc, 'edx-enterprise', 'enterprise Documentation',
-     author, 'edx-enterprise', 'Your project description goes here',
-     'Miscellaneous'),
+    ('index', 'Open edX Enterprise Service', u'Open edX Enterprise Service Documentation',
+     u'edX', 'Open edX Enterprise Service', 'Open edX Enterprise Service',
+     'Miscellaneous'
```
Comment (e-kolpakov) [inline docs/conf.py]: @mattdrayer Same idea - please use constants instead of plain strings where possible

File: docs/conf.py [hunk 9]
```diff
@@ -136,349 +106,163 @@
...
-    'django': ('https://docs.djangoproject.com/en/1.10/', 'https://docs.djangoproject.com/en/1.10/_objects/'),
-    'model_utils': ('https://django-model-utils.readthedocs.io/en/latest/', None),
-}
-
-
-def on_init(app):  # pylint: disable=unused-argument
-    """
-    Run sphinx-apidoc after Sphinx initialization.
-
-    Read the Docs won't run tox or custom shell commands, so we need this to
```
Comment (e-kolpakov) [inline docs/conf.py]: @mattdrayer I'm curious how *well* it works without it. If I understand right `autodoc` plugin generates `enterprise.rst` and `modules.rst` files that aren't checked in (and thus not land automatically at RTD) and this method generate them there by running `autodoc`. Could you please make sure you can access all docs pages on RTD, especially "enterprise" or "enterprise package".

**Pull request closed without merging**